fanficfare/adapters/adapter_asianfanficscom.py

```diff
@@ -113,6 +113,23 @@ def doAdultCheck(self, url, soup):
         else:
             return False
 
+    def doSubCheck(self, url, soup):
+        check = soup.find('div',{'class':'click-to-read-full'})
+        if check:
+            logger.debug("Subscription required to get all HTML tags")
+            #does not work when using https - 403
+            subUrl = 'http://' + self.getSiteDomain() + soup.find('a',{'id':'subscribe'})['href']
+            self._fetchUrl(subUrl)
+            data = self._fetchUrl(url,usecache=False)
+            soup = self.make_soup(data)
+            check = soup.find('div',{'class':'click-to-read-full'})
+            if check:
+                raise exceptions.FailedToDownload("Error downloading Chapter: %s!  Missing required element!" % url)
```

Comment:
Throwing a `FailedToDownload` here seems rather extreme.  Sure you don't want to fall back to using the value already obtained?

Reply:
I withdraw this objection--or at least change it--subscription required stories leave out parts of the text, not just formatting tags, so a `FailedToDownload` is appropriate.  But the message should be something about the story requiring subscription.